### PR TITLE
WIP: Added JacksonSupport

### PR DESCRIPTION
**Pull request by ljnelson**
Signed-off-by: Laird Nelson <ljnelson@gmail.com>

I put this together following the existing `JsonSupport` implementation.  All comments welcomed.  See also: #45.

### CODE REVIEW

**Comment by ljnelson** [comment]
The pipeline failure seems unrelated and intermittent; @romain-grecourt can you cast a set of eyeballs on it and confirm at your leisure?

**Comment by romain-grecourt** [comment]
I see `Caused by: java.lang.NoClassDefFoundError: com/fasterxml/jackson/annotation/JsonMerge`, this could be related to a different version of jackson being pulled in with your changes.

**Comment by ljnelson** [comment]
Hate to say it, but works fine on my machine with a clean checkout?

**Comment by romain-grecourt** [comment]
let's retry the pipeline build job and let's see if that's intermittent.

**Comment by ljnelson** [comment]
Found the issue.  The `config/etcd` Maven module uses `etcd4j` which until recently pulls in `jackson-databind` and `jackson-annotations` and `jackson-module-afterburner` at 2.8.11.  We have set our `${version.lib.jackson}` variable to 2.9.4, but only mentioned `jackson-core` in `<dependencyManagement>`.  We need to make sure we mention everything Jackson-related throughout all of Helidon explicitly in the `<dependencyManagement>` section as well, since Jackson doesn't ship a truly inclusive BOM.  I'll try to do that in this PR.

**Comment by ljnelson** [comment]
@barchetta I'd be interested in your thoughts particularly with regard to the changes in the root `pom.xml`.

File: webserver/jackson/src/main/java/io/helidon/webserver/jackson/JacksonSupport.java [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+     * Creates a new {@link JacksonSupport}.
+     *
+     * @param objectMapperProvider a {@link BiFunction} that returns
+     * an {@link ObjectMapper} when given a {@link ServerRequest} and
+     * a {@link ServerResponse}; must not be {@code null}
+     *
+     * @exception NullPointerException if {@code objectMapperProvider}
+     * is {@code null}
+     */
+    @SuppressWarnings("checkstyle:linelength")
```
Comment (tomas-langer) [inline webserver/jackson/src/main/java/io/helidon/webserver/jackson/JacksonSupport.java]: This warning should never be suppressed. Please format differently.

Comment (ljnelson) [reply webserver/jackson/src/main/java/io/helidon/webserver/jackson/JacksonSupport.java]: Done

**Comment by romain-grecourt** [comment]
@ljnelson @tomas-langer has merged pr #319 for the media type refactoring, can you can go ahead and refresh this PR ?

**Comment by ljnelson** [comment]
This PR is now a mess; I'm going to close it, delete my branch, and redo it; rebasing this is a nightmare.

**Comment by ljnelson** [comment]
Closing per earlier comments.

**Pull request closed without merging**